### ISSUE TITLE
[Feature]: Flatpak

### ISSUE THREAD

**Issue by aswerkljh** [KSP-CKAN/CKAN#4098]
### Problem

apt repos are annoying and can cause problems down the line
.deb files are janky
flatpaks are amazing and Just Works™️

### Suggestion

It would be amazing and honestly downright incredible if you would consider shipping flatpaks, as it circumvents a lot of trouble with apt repos and .deb files. 

Thank you

**Comment by HebaruSan** [comment]
See #2636 for previous discussion on a similar idea.

**Comment by HebaruSan** [comment]
> apt repos are annoying and can cause problems down the line .deb files are janky
> a lot of trouble with apt repos and .deb files.

@aswerkljh, can you put some meat on these bonds? What problems have you had with APT and deb packges?
Even if we did add more package types, the old ones wouldn't go away, so we'd want to fix any problems they had.

**Comment by aswerkljh** [comment]
>  so we'd want to fix any problems they had.

I don't mean to sound rude, but I think fixing dependency hell is beyond the scope of CKAN :stuck_out_tongue: 

I've been using your apt repo and it's nice that it works but it means I have to trust you. I have to trust that you don't add a funky package to your repo which is a version ahead of what I have in my main repos which is gonna toss my system into a loop. Adding new repos with appropriate security is also not totally straight forward and managing it after the fact is absolutely not straight forward. 

Deb packages are just outdated, frankly. I would rather not use a piece of software than have to `--fix-broken` again. Anytime a user has to whip out `dpkg`, something dumb is going on.

Flatpaks have none of these issues

**Comment by HebaruSan** [comment]
OK, so it's more about hypothetical/potential problems than actual issues with our APT repo? Sounds like everything is actually working fine?

**Comment by aswerkljh** [comment]
I don't believe I ever, once, said your repos or deb were inherently broken. I said apt repos and deb packages are unwieldy to work with in comparison to a flatpak.

**Comment by HebaruSan** [comment]
I thought "janky" and "a lot of trouble" could have been read either way, hence the follow-up questions. Thanks for answering.

**Comment by techman83** [comment]
@aswerkljh - there is some merit to tools like AppImage/Flatpack, when you are dealing with complex applications with complex dependencies. This is not the case for CKAN, it works pretty universally across mono versions and has no complicated dependency chain.

It would be yet another process for us to maintain, that isn't fundamentally solving a problem that exists for the project. Though we are always happy for robust contributions :slightly_smiling_face: 

On a personal note, I very much prefer things that are distributed via the system package manager. I really don't want to go either manually acquiring updates, or maintaining multiple package managers. In my experience it's pretty rare to encounter issues with apt, unless you're doing something very bizarre. And the PPA ecosystem + upgrade process for them (at least on Linux mint, no idea about Ubuntu and their drive for Yet Another Package Manager), is pretty seamless these days.

**Comment by AltSimon** [comment]
Another advantage of flatpaks are for immutable distros like my current distro, Fedora Kinoite where installing packages directly on the system is not recommended because it makes the system a bit slower and takes gigabytes of disk space. The best option right now is to use a container like distrobox.
Maybe [this](https://github.com/NickvisionApps/FlatpakGenerator) can be used, but I really don't know

**Comment by Int-Circuit** [comment]
> Another advantage of flatpaks are for immutable distros like my current distro, Fedora Kinoite where installing packages directly on the system is not recommended because it makes the system a bit slower and takes gigabytes of disk space. The best option right now is to use a container like distrobox. Maybe [this](https://github.com/NickvisionApps/FlatpakGenerator) can be used, but I really don't know

I took a look at that tool but sadly it's just a flatpak builder wrapper. To use it, builder expects 4 GUI toolkits:

    Prerequisites
        The application is built with a Linux-compatible .NET desktop application framework such as:

            [Avalonia UI](https://avaloniaui.net/)

            [Uno Platform](https://platform.uno/)

            [Eto](https://github.com/picoe/Eto)

            [GTKSharp](https://github.com/GtkSharp/GtkSharp)

            [Gir.Core](https://gircore.github.io/)

    The application’s source code is hosted on a Git server such as GitHub, GitLab, or Bitbucket


see https://docs.flatpak.org/en/latest/dotnet.html. So before we can build it, we need to migrate to one of those UIs (which I cannot give an ETA).
